cmake/Modules/FindXCCL.cmake

```diff
@@ -0,0 +1,70 @@
+# This will define the following variables:
+# XCCL_FOUND               : True if the system has the XCCL library.
+# XCCL_INCLUDE_DIR         : Include directories needed to use XCCL.
+# XCCL_LIBRARY_DIR         ：The path to the XCCL library.
+# XCCL_LIBRARY             : XCCL library fullname.
+
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+
+set(XCCL_ROOT "/opt/intel/oneapi/ccl/latest")
```

Comment:
Should we check the OS type first? If on Windows, what we should do, assert? warning?